sdk/storage/azure-storage-blob/azure/storage/blob/models.py

```diff
@@ -879,58 +864,35 @@ class ContainerPermissions(object):
         delete a container with a container SAS. Use an account SAS instead.
     :param bool list:
         List blobs in the container.
-    :param str _str:
-        A string representing the permissions.
     """
-
-    DELETE = None  # type: ContainerPermissions
-    LIST = None  # type: ContainerPermissions
-    READ = None  # type: ContainerPermissions
-    WRITE = None  # type: ContainerPermissions
-
-    def __init__(self, read=False, write=False, delete=False, list=False, _str=None):  # pylint: disable=redefined-builtin
-        if not _str:
-            _str = ''
-        self.read = read or ('r' in _str)
-        self.write = write or ('w' in _str)
-        self.delete = delete or ('d' in _str)
-        self.list = list or ('l' in _str)
-
-    def __or__(self, other):
-        return ContainerPermissions(_str=str(self) + str(other))
-
-    def __add__(self, other):
-        return ContainerPermissions(_str=str(self) + str(other))
+    def __init__(self, read=False, write=False, delete=False, list=False):  # pylint: disable=redefined-builtin
+        self.read = read
+        self.write = write
+        self.delete = delete
+        self.list = list
+        self._str = (('r' if self.read else '') +
+                     ('w' if self.write else '') +
+                     ('d' if self.delete else '') +
+                     ('l' if self.list else ''))
 
     def __str__(self):
-        return (('r' if self.read else '') +
-                ('w' if self.write else '') +
-                ('d' if self.delete else '') +
-                ('l' if self.list else ''))
-
-
-ContainerPermissions.DELETE = ContainerPermissions(delete=True)
-ContainerPermissions.LIST = ContainerPermissions(list=True)
-ContainerPermissions.READ = ContainerPermissions(read=True)
-ContainerPermissions.WRITE = ContainerPermissions(write=True)
-
+        return self._str
 
-class BlobPermissions(object):
-    """BlobPermissions class to be used with
+    @classmethod
+    def from_string(cls, permission):
+        p_read = 'r' in permission
+        p_write = 'w' in permission
+        p_delete = 'd' in permission
+        p_list = 'l' in permission
+        parsed = cls(p_read, p_write, p_delete, p_list)
+        parsed._str = permission # pylint: disable = protected-access
```

Comment:
same question as the the on in AccountSasPermission

Reply:
no, the parsed._str will be 'abcd' for non-lossy behavior. (We want to keep the original string.)